Fix for blacklist ids bug and added pagination

powahftw opened the pull request.

powahftw committed:
provide default empty list for blacklist ids

powahftw committed:
added paginations to /logs page

mvkv commented on static/css/style.css:
```diff
@@ -318,6 +318,17 @@ ul {
   margin: 0;
 }
 
+.pagination-container button {
```
Add cursor pointer and change the colors of the arrows to white

mvkv replied:
Done

mvkv commented on flask_server.py:
```diff
@@ -208,9 +209,13 @@ def logout():
+def get_logs(page):
+    logs = get_app_logs()
+    paginated_logs = [logs[start:start+PAGINATE_EVERY_N_ROWS] for start in range(0, len(logs), PAGINATE_EVERY_N_ROWS)]
+    if page < 1 or page > len(paginated_logs):
+        return jsonify({"page": -1, "max_page": len(paginated_logs), "logs": []})
```
I think these two lines are useless

mvkv replied:
Resolved client side

mvkv commented on static/js/logs.js:
```diff
@@ -12,9 +13,34 @@ const renderLogs = (logLines) => {
+    return; // Don't display pagination on log files that fit in a single page.
+  }
+  const root = document.createElement('div');
+  if (pageN > 1) {
+    decrease = document.createElement('button');
+    decrease.innerText = '<';
+    decrease.addEventListener('click', () => getAndRenderLogs(pageN - 1));
+    root.append(decrease);
+  }
+  root.append(document.createTextNode(`${pageN} / ${maxPage}`));
+  if (pageN < maxPage) {
+    increase = document.createElement('button');
+    increase.innerText = '>';
+    increase.addEventListener('click', () => getAndRenderLogs(pageN + 1));
+    root.append(increase);
+  }
+  paginationNode.append(root);
```
nit: fix the spacing when there's only one arrow

mvkv replied:
Done

mvkv committed:
Fixed spacing of pagination buttons in logs,
fixed empty logs pagination logic,
added cursor and color to pagination buttons

mvkv committed:
Added unchanged warning on text inputs

powahftw commented on static/js/settings.js:
```diff
@@ -189,6 +189,11 @@ const setUpButtonsListeners = () => {
   const submitBtn = document.getElementById('submitBtn');
   const addExtraIDBtn = document.getElementById('add-extra-id');
   const addBlacklistedIDBtn = document.getElementById('add-blacklisted-id');
+  const changingInputs = [
```
Move this below, right above where you use them, so it's logically closer. You could also just store the id and add the document.getElementById in the loop.

powahftw commented on static/js/logs.js:
```diff
@@ -15,9 +15,9 @@ const renderLogs = (logLines) => {
+  if (maxPage <= 1) {
     return; // Don't display pagination on log files that fit in a single page.
   }
   const root = document.createElement('div');
```
Move the buttons in the HTML if they are always rendered. Add an id (previous-page, next-page) to attach the event listener. Make them invisible by default.

mvkv committed:
Moved changingInputs variable near his logic
